tests/test_exceptions.cc

```diff
@@ -66,3 +66,25 @@ TEST_CASE("Error: Copy assignment", "[exceptions]")
     REQUIRE(e2.what() == "Test"s);
     REQUIRE(e2.get_index() == e.get_index());
 }
+
+TEST_CASE("Error: operator==", "[exceptions]")
+{
+    REQUIRE(Error("Test", 42) == Error("Test", 42));
+    REQUIRE(Error("Test", gul14::nullopt) == Error("Test", gul14::nullopt));
+    REQUIRE(Error("", 42) == Error("", 42));
+
+    REQUIRE_FALSE(Error("test", 42) == Error("Test", 42));
```

Comment:
I would have added such a more obvious/simple test:
```suggestion
    REQUIRE_FALSE(Error("test", 42) == Error("Test", 42));
    REQUIRE_FALSE(Error("test", 42) == Error("test", 23));
```

And in principle I would not have used `test` and `Test` because it is very hard to see the difference, but like `test` and `TEST` or `One` and `Two`.

Reply:
Good idea, I changed the tests a bit according to your proposal.